rlp/src/impls.rs

```diff
@@ -207,20 +208,34 @@ macro_rules! impl_decodable_for_hash {
 	}
 }
 
+#[cfg(feature = "ethereum")]
 impl_encodable_for_hash!(H64);
```

Comment:
can we move all of these macro calls into a separate module to avoid writing `cfg` 20 times? :)